Frontend/src/app/home/filter-and-restaurants/filter-and-restaurants.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Input } from '@angular/core';
+import { SearchAndFilterComponent } from '../search-and-filter/search-and-filter.component';
+import { NewRestaurantsCarouselComponent } from '../new-restaurants-carousel/new-restaurants-carousel.component';
+import { Restaurant } from '../../model/restaurant.model';
+import { RestaurantCardComponent } from '../../restaurant-card/restaurant-card.component';
+import { FilterData } from '../../model/filter-data.model';
+import { log } from '@angular-devkit/build-angular/src/builders/ssr-dev-server';
+
+@Component({
+	selector: 'app-filter-and-restaurants',
+	standalone: true,
+	imports: [
+		SearchAndFilterComponent,
+		NewRestaurantsCarouselComponent,
+		RestaurantCardComponent,
+	],
+	templateUrl: './filter-and-restaurants.component.html',
+	styleUrl: './filter-and-restaurants.component.scss',
+})
+export class FilterAndRestaurantsComponent {
+	protected _allRestaurants: Restaurant[] = [];
+	protected filteredRestaurants: Restaurant[] = [];
+	private filterData: FilterData;
```

Comment:
Can't you make a setter for filterData, and make it an Input property in the other component as I mentioned? That way, you don't have to worry about emitting an event every time it's changed. Then you can just call the `onFilter` method in the setter. (If the change detector complies, but calling `detectChanges` might be better?)